modules/internal/ChapelRange.chpl

```diff
@@ -18,52 +18,86 @@
  */
 /*
   A ``range`` is a first-class, constant-space representation of a
-  regular sequence of integer indices. Ranges support iteration over the
-  sequences they represent as well as operations such as slicing, shifting,
-  comparisons, striding, counting and aligning.
+  regular sequence of values.  These values are typically integers,
+  though ranges over enumerated types are also supported.  Ranges
+  support iteration over the sequences they represent as well as
+  operations such as counting, striding, intersection, shifting, and
+  comparisons.
+
+  Range Values
+  ------------
+  In their simplest form, ranges are represented by their low and high
+  bounds:
 
-  Range Construction Operations:
-  New ranges can be constructed using the striding, counting, and alignment
-  operators, ``by``, ``#`` and ``align``
 
   .. code-block:: chapel
 
-    0..#10 // 0, 1, 2, 3, 4, 5, 6, 7, 8, 9
-    0..10 by 2 // 0, 2, 4, 6, 8, 10
-    0..10 by 2 align 1 // 1, 3, 5, 7, 9
+    1..3    // 1, 2, 3
+    0..n    // 0, 1, 2, 3, ..., n
+    lo..hi  // lo, lo+1, lo+2, ..., hi
 
-  Range Slicing:
-  A range can be sliced with another range to form a new range that is the intersection of the two ranges.
+  Ranges may also be `unbounded`, in which case, the lower and/or upper
+  bounds may be omitted:
 
   .. code-block:: chapel
 
-    (1..10)(3..8) // 3..8
-    (0..20)(1..20 by 2) // 1..20 by 2
-    (1..10)(5..) // 5..10
-    (1..10)(..5) // 1..5
+    1..   // 1, 2, 3, ...
+    ..10  // .., 8, 9, 10
+    ..    // ..., -2, -1, 0, 1, 2, ...
 
-  Range Shifting:
-  A range can be shifted by an integer using the ``+`` and ``-`` operators.
+  Ranges over enumerated types respect the declaration order of its values:
 
   .. code-block:: chapel
 
-    (1..10) + 5 // 6..15
-    (1..10) - 3 // -2..7
-    (1..) + 1 // 2..
-    (..10) + 1 // ..11
+    enum color {red, orange, yellow, green, blue, indigo, violet};
+    color.orange..color.green;   // orange, yellow, green
```

Comment:
That's a good point.  I'm not wild about that specific example since it contains duplicate values and there's some question as to whether or not we want to continue to support that (issue #9786).  Let me see if I can come up with something else.

Reply:
I ended up going with the same color set (since it's used throughout the docs) and gave them a permutation of the values 1..7 indicating how "bright" the color is (though I didn't describe this... it was just my mental model).